File: r/R/V3ResultStop.r
# PTV Timetable API - Version 3
#
# The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
#
# OpenAPI spec version: v3
# 
# Generated by: https://github.com/swagger-api/swagger-codegen.git

#' V3ResultStop Class
#'
#' @field stop_distance 
#' @field stop_suburb 
#' @field route_type 
#' @field routes 
#' @field stop_latitude 
#' @field stop_longitude 
#' @field stop_sequence 
#' @field stop_id 
#' @field stop_name 
#' @field stop_landmark 
#'
#' @importFrom R6 R6Class
#' @importFrom jsonlite fromJSON toJSON
#' @export
V3ResultStop <- R6::R6Class(
  'V3ResultStop',
  public = list(
    `stop_distance` = NULL,
    `stop_suburb` = NULL,
    `route_type` = NULL,
    `routes` = NULL,
    `stop_latitude` = NULL,
    `stop_longitude` = NULL,
    `stop_sequence` = NULL,
    `stop_id` = NULL,
    `stop_name` = NULL,
    `stop_landmark` = NULL,
    initialize = function(`stop_distance`, `stop_suburb`, `route_type`, `routes`, `stop_latitude`, `stop_longitude`, `stop_sequence`, `stop_id`, `stop_name`, `stop_landmark`){
      if (!missing(`stop_distance`)) {
        stopifnot(is.numeric(`stop_distance`), length(`stop_distance`) == 1)
        self$`stop_distance` <- `stop_distance`
      }
      if (!missing(`stop_suburb`)) {
        stopifnot(is.character(`stop_suburb`), length(`stop_suburb`) == 1)
        self$`stop_suburb` <- `stop_suburb`
      }
      if (!missing(`route_type`)) {
        stopifnot(is.numeric(`route_type`), length(`route_type`) == 1)
        self$`route_type` <- `route_type`
      }
      if (!missing(`routes`)) {
        stopifnot(is.list(`routes`), length(`routes`) != 0)
        lapply(`routes`, function(x) stopifnot(R6::is.R6(x)))
        self$`routes` <- `routes`
      }
      if (!missing(`stop_latitude`)) {
        stopifnot(is.numeric(`stop_latitude`), length(`stop_latitude`) == 1)
        self$`stop_latitude` <- `stop_latitude`
      }
      if (!missing(`stop_longitude`)) {
        stopifnot(is.numeric(`stop_longitude`), length(`stop_longitude`) == 1)
        self$`stop_longitude` <- `stop_longitude`
      }
      if (!missing(`stop_sequence`)) {
        stopifnot(is.numeric(`stop_sequence`), length(`stop_sequence`) == 1)
        self$`stop_sequence` <- `stop_sequence`
      }
      if (!missing(`stop_id`)) {
        stopifnot(is.numeric(`stop_id`), length(`stop_id`) == 1)
        self$`stop_id` <- `stop_id`
      }
      if (!missing(`stop_name`)) {
        stopifnot(is.character(`stop_name`), length(`stop_name`) == 1)
        self$`stop_name` <- `stop_name`
      }
      if (!missing(`stop_landmark`)) {
        stopifnot(is.character(`stop_landmark`), length(`stop_landmark`) == 1)
        self$`stop_landmark` <- `stop_landmark`
      }
    },
    toJSON = function() {
      V3ResultStopObject <- list()
      if (!is.null(self$`stop_distance`)) {
        V3ResultStopObject[['stop_distance']] <- self$`stop_distance`
      }
      if (!is.null(self$`stop_suburb`)) {
        V3ResultStopObject[['stop_suburb']] <- self$`stop_suburb`
      }
      if (!is.null(self$`route_type`)) {
        V3ResultStopObject[['route_type']] <- self$`route_type`
      }
      if (!is.null(self$`routes`)) {
        V3ResultStopObject[['routes']] <- lapply(self$`routes`, function(x) x$toJSON())
      }
      if (!is.null(self$`stop_latitude`)) {
        V3ResultStopObject[['stop_latitude']] <- self$`stop_latitude`
      }
      if (!is.null(self$`stop_longitude`)) {
        V3ResultStopObject[['stop_longitude']] <- self$`stop_longitude`
      }
      if (!is.null(self$`stop_sequence`)) {
        V3ResultStopObject[['stop_sequence']] <- self$`stop_sequence`
      }
      if (!is.null(self$`stop_id`)) {
        V3ResultStopObject[['stop_id']] <- self$`stop_id`
      }
      if (!is.null(self$`stop_name`)) {
        V3ResultStopObject[['stop_name']] <- self$`stop_name`
      }
      if (!is.null(self$`stop_landmark`)) {
        V3ResultStopObject[['stop_landmark']] <- self$`stop_landmark`
      }

      V3ResultStopObject
    },
    fromJSON = function(V3ResultStopJson) {
      V3ResultStopObject <- jsonlite::fromJSON(V3ResultStopJson)
      if (!is.null(V3ResultStopObject$`stop_distance`)) {
        self$`stop_distance` <- V3ResultStopObject$`stop_distance`
      }
      if (!is.null(V3ResultStopObject$`stop_suburb`)) {
        self$`stop_suburb` <- V3ResultStopObject$`stop_suburb`
      }
      if (!is.null(V3ResultStopObject$`route_type`)) {
        self$`route_type` <- V3ResultStopObject$`route_type`
      }
      if (!is.null(V3ResultStopObject$`routes`)) {
        self$`routes` <- lapply(V3ResultStopObject$`routes`, function(x) {
          routesObject <- V3ResultRoute$new()
          routesObject$fromJSON(jsonlite::toJSON(x, auto_unbox = TRUE))
          routesObject
        })
      }
      if (!is.null(V3ResultStopObject$`stop_latitude`)) {
        self$`stop_latitude` <- V3ResultStopObject$`stop_latitude`
      }
      if (!is.null(V3ResultStopObject$`stop_longitude`)) {
        self$`stop_longitude` <- V3ResultStopObject$`stop_longitude`
      }
      if (!is.null(V3ResultStopObject$`stop_sequence`)) {
        self$`stop_sequence` <- V3ResultStopObject$`stop_sequence`
      }
      if (!is.null(V3ResultStopObject$`stop_id`)) {
        self$`stop_id` <- V3ResultStopObject$`stop_id`
      }
      if (!is.null(V3ResultStopObject$`stop_name`)) {
        self$`stop_name` <- V3ResultStopObject$`stop_name`
      }
      if (!is.null(V3ResultStopObject$`stop_landmark`)) {
        self$`stop_landmark` <- V3ResultStopObject$`stop_landmark`
      }
    },
    toJSONString = function() {
       sprintf(
        '{
           "stop_distance": %d,
           "stop_suburb": %s,
           "route_type": %d,
           "routes": [%s],
           "stop_latitude": %d,
           "stop_longitude": %d,
           "stop_sequence": %d,
           "stop_id": %d,
           "stop_name": %s,
           "stop_landmark": %s
        }',
        self$`stop_distance`,
        self$`stop_suburb`,
        self$`route_type`,
        lapply(self$`routes`, function(x) paste(x$toJSON(), sep=",")),
        self$`stop_latitude`,
        self$`stop_longitude`,
        self$`stop_sequence`,
        self$`stop_id`,
        self$`stop_name`,
        self$`stop_landmark`
      )
    },
    fromJSONString = function(V3ResultStopJson) {
      V3ResultStopObject <- jsonlite::fromJSON(V3ResultStopJson)
      self$`stop_distance` <- V3ResultStopObject$`stop_distance`
      self$`stop_suburb` <- V3ResultStopObject$`stop_suburb`
      self$`route_type` <- V3ResultStopObject$`route_type`
      self$`routes` <- lapply(V3ResultStopObject$`routes`, function(x) V3ResultRoute$new()$fromJSON(jsonlite::toJSON(x, auto_unbox = TRUE)))
      self$`stop_latitude` <- V3ResultStopObject$`stop_latitude`
      self$`stop_longitude` <- V3ResultStopObject$`stop_longitude`
      self$`stop_sequence` <- V3ResultStopObject$`stop_sequence`
      self$`stop_id` <- V3ResultStopObject$`stop_id`
      self$`stop_name` <- V3ResultStopObject$`stop_name`
      self$`stop_landmark` <- V3ResultStopObject$`stop_landmark`
    }
  )
)

File: swagger/php/SwaggerClient-php/lib/Model/V3ResultStop.php
<?php
/**
 * V3ResultStop
 *
 * PHP version 5
 *
 * @category Class
 * @package  Swagger\Client
 * @author   Swagger Codegen team
 * @link     https://github.com/swagger-api/swagger-codegen
 */

/**
 * PTV Timetable API - Version 3
 *
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.
 *
 * OpenAPI spec version: v3
 * 
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 * Swagger Codegen version: 2.4.39
 */

/**
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen
 * Do not edit the class manually.
 */

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

/**
 * V3ResultStop Class Doc Comment
 *
 * @category Class
 * @package  Swagger\Client
 * @author   Swagger Codegen team
 * @link     https://github.com/swagger-api/swagger-codegen
 */
class V3ResultStop implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    /**
      * The original name of the model.
      *
      * @var string
      */
    protected static $swaggerModelName = 'V3.ResultStop';

    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $swaggerTypes = [
        'stop_distance' => 'float',
        'stop_suburb' => 'string',
        'route_type' => 'int',
        'routes' => '\Swagger\Client\Model\V3ResultRoute[]',
        'stop_latitude' => 'float',
        'stop_longitude' => 'float',
        'stop_sequence' => 'int',
        'stop_id' => 'int',
        'stop_name' => 'string',
        'stop_landmark' => 'string'
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $swaggerFormats = [
        'stop_distance' => 'float',
        'stop_suburb' => null,
        'route_type' => 'int32',
        'routes' => null,
        'stop_latitude' => 'float',
        'stop_longitude' => 'float',
        'stop_sequence' => 'int32',
        'stop_id' => 'int32',
        'stop_name' => null,
        'stop_landmark' => null
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @var string[]
     */
    protected static $attributeMap = [
        'stop_distance' => 'stop_distance',
        'stop_suburb' => 'stop_suburb',
        'route_type' => 'route_type',
        'routes' => 'routes',
        'stop_latitude' => 'stop_latitude',
        'stop_longitude' => 'stop_longitude',
        'stop_sequence' => 'stop_sequence',
        'stop_id' => 'stop_id',
        'stop_name' => 'stop_name',
        'stop_landmark' => 'stop_landmark'
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'stop_distance' => 'setStopDistance',
        'stop_suburb' => 'setStopSuburb',
        'route_type' => 'setRouteType',
        'routes' => 'setRoutes',
        'stop_latitude' => 'setStopLatitude',
        'stop_longitude' => 'setStopLongitude',
        'stop_sequence' => 'setStopSequence',
        'stop_id' => 'setStopId',
        'stop_name' => 'setStopName',
        'stop_landmark' => 'setStopLandmark'
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'stop_distance' => 'getStopDistance',
        'stop_suburb' => 'getStopSuburb',
        'route_type' => 'getRouteType',
        'routes' => 'getRoutes',
        'stop_latitude' => 'getStopLatitude',
        'stop_longitude' => 'getStopLongitude',
        'stop_sequence' => 'getStopSequence',
        'stop_id' => 'getStopId',
        'stop_name' => 'getStopName',
        'stop_landmark' => 'getStopLandmark'
    ];

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @return array
     */
    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * The original name of the model.
     *
     * @return string
     */
    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    

    

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     *                      initializing the model
     */
    public function __construct(array $data = null)
    {
        $this->container['stop_distance'] = isset($data['stop_distance']) ? $data['stop_distance'] : null;
        $this->container['stop_suburb'] = isset($data['stop_suburb']) ? $data['stop_suburb'] : null;
        $this->container['route_type'] = isset($data['route_type']) ? $data['route_type'] : null;
        $this->container['routes'] = isset($data['routes']) ? $data['routes'] : null;
        $this->container['stop_latitude'] = isset($data['stop_latitude']) ? $data['stop_latitude'] : null;
        $this->container['stop_longitude'] = isset($data['stop_longitude']) ? $data['stop_longitude'] : null;
        $this->container['stop_sequence'] = isset($data['stop_sequence']) ? $data['stop_sequence'] : null;
        $this->container['stop_id'] = isset($data['stop_id']) ? $data['stop_id'] : null;
        $this->container['stop_name'] = isset($data['stop_name']) ? $data['stop_name'] : null;
        $this->container['stop_landmark'] = isset($data['stop_landmark']) ? $data['stop_landmark'] : null;
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    /**
     * Gets stop_distance
     *
     * @return float
     */
    public function getStopDistance()
    {
        return $this->container['stop_distance'];
    }

    /**
     * Sets stop_distance
     *
     * @param float $stop_distance Distance of stop from input location (in metres); returns 0 if no location is input
     *
     * @return $this
     */
    public function setStopDistance($stop_distance)
    {
        $this->container['stop_distance'] = $stop_distance;

        return $this;
    }

    /**
     * Gets stop_suburb
     *
     * @return string
     */
    public function getStopSuburb()
    {
        return $this->container['stop_suburb'];
    }

    /**
     * Sets stop_suburb
     *
     * @param string $stop_suburb suburb of stop
     *
     * @return $this
     */
    public function setStopSuburb($stop_suburb)
    {
        $this->container['stop_suburb'] = $stop_suburb;

        return $this;
    }

    /**
     * Gets route_type
     *
     * @return int
     */
    public function getRouteType()
    {
        return $this->container['route_type'];
    }

    /**
     * Sets route_type
     *
     * @param int $route_type Transport mode identifier
     *
     * @return $this
     */
    public function setRouteType($route_type)
    {
        $this->container['route_type'] = $route_type;

        return $this;
    }

    /**
     * Gets routes
     *
     * @return \Swagger\Client\Model\V3ResultRoute[]
     */
    public function getRoutes()
    {
        return $this->container['routes'];
    }

    /**
     * Sets routes
     *
     * @param \Swagger\Client\Model\V3ResultRoute[] $routes List of routes travelling through the stop
     *
     * @return $this
     */
    public function setRoutes($routes)
    {
        $this->container['routes'] = $routes;

        return $this;
    }

    /**
     * Gets stop_latitude
     *
     * @return float
     */
    public function getStopLatitude()
    {
        return $this->container['stop_latitude'];
    }

    /**
     * Sets stop_latitude
     *
     * @param float $stop_latitude Geographic coordinate of latitude at stop
     *
     * @return $this
     */
    public function setStopLatitude($stop_latitude)
    {
        $this->container['stop_latitude'] = $stop_latitude;

        return $this;
    }

    /**
     * Gets stop_longitude
     *
     * @return float
     */
    public function getStopLongitude()
    {
        return $this->container['stop_longitude'];
    }

    /**
     * Sets stop_longitude
     *
     * @param float $stop_longitude Geographic coordinate of longitude at stop
     *
     * @return $this
     */
    public function setStopLongitude($stop_longitude)
    {
        $this->container['stop_longitude'] = $stop_longitude;

        return $this;
    }

    /**
     * Gets stop_sequence
     *
     * @return int
     */
    public function getStopSequence()
    {
        return $this->container['stop_sequence'];
    }

    /**
     * Sets stop_sequence
     *
     * @param int $stop_sequence Sequence of the stop on the route/run; return 0 when route_id or run_id not specified. Order ascendingly by this field (when non zero) to get physical order (earliest first) of stops on the route_id/run_id.
     *
     * @return $this
     */
    public function setStopSequence($stop_sequence)
    {
        $this->container['stop_sequence'] = $stop_sequence;

        return $this;
    }

    /**
     * Gets stop_id
     *
     * @return int
     */
    public function getStopId()
    {
        return $this->container['stop_id'];
    }

    /**
     * Sets stop_id
     *
     * @param int $stop_id Stop identifier
     *
     * @return $this
     */
    public function setStopId($stop_id)
    {
        $this->container['stop_id'] = $stop_id;

        return $this;
    }

    /**
     * Gets stop_name
     *
     * @return string
     */
    public function getStopName()
    {
        return $this->container['stop_name'];
    }

    /**
     * Sets stop_name
     *
     * @param string $stop_name Name of stop
     *
     * @return $this
     */
    public function setStopName($stop_name)
    {
        $this->container['stop_name'] = $stop_name;

        return $this;
    }

    /**
     * Gets stop_landmark
     *
     * @return string
     */
    public function getStopLandmark()
    {
        return $this->container['stop_landmark'];
    }

    /**
     * Sets stop_landmark
     *
     * @param string $stop_landmark Landmark in proximity of stop
     *
     * @return $this
     */
    public function setStopLandmark($stop_landmark)
    {
        $this->container['stop_landmark'] = $stop_landmark;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param integer $offset Offset
     *
     * @return boolean
     */
    #[\ReturnTypeWillChange]
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param integer $offset Offset
     *
     * @return mixed
     */
    #[\ReturnTypeWillChange]
    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    /**
     * Sets value based on offset.
     *
     * @param integer $offset Offset
     * @param mixed   $value  Value to be set
     *
     * @return void
     */
    #[\ReturnTypeWillChange]
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param integer $offset Offset
     *
     * @return void
     */
    #[\ReturnTypeWillChange]
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    /**
     * Gets the string presentation of the object
     *
     * @return string
     */
    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}


